MantidPlot/src/Mantid/MantidDock.cpp

```diff
@@ -1737,38 +1936,27 @@ QStringList MantidTreeWidget::getSelectedWorkspaceNames() const
 }
 
 /**
-* Allows users to choose spectra from the selected workspaces by presenting them
-* with a dialog box.  Skips showing the dialog box and automatically chooses
-* workspace index 0 for all selected workspaces if one or more of the them are
-* single-spectrum workspaces.
-*
-* We also must filter the list of selected workspace names to account for any
-* non-MatrixWorkspaces that may have been selected.  In particular WorkspaceGroups
-* (the children of which are to be included if they are MatrixWorkspaces) and
-* TableWorkspaces (which are implicitly excluded).  We only want workspaces we
-* can actually plot!
-*
-* @param showWaterfallOpt If true, show the waterfall option on the dialog
-* @return :: A MantidWSIndexDialog::UserInput structure listing the selected options
-*/
-MantidWSIndexDialog::UserInput MantidTreeWidget::chooseSpectrumFromSelected(bool showWaterfallOpt) const
-{
+ * Filter the list of selected workspace names to account for any
+ * non-MatrixWorkspaces that may have been selected.  In particular
+ * WorkspaceGroups (the children of which are to be included if they are
+ * MatrixWorkspaces) and TableWorkspaces (which are implicitly excluded).
+ * We only want workspaces we can actually plot!
+ */
+QList<MatrixWorkspace_const_sptr>
+MantidTreeWidget::getSelectedWorkspaces() const {
```

Comment:
The name of this function does not imply that certain workspaces will be omitted from the final list.